openupgrade_scripts/scripts/account/17.0.1.2/post-migration.py

```diff
@@ -0,0 +1,535 @@
+        "tax_receivable_account_id",
+        origin_id_column,
+    )
+    convert_from_company_dependent(
+        env,
+        "account.tax.group",
+        "property_tax_payable_account_id",
+        "tax_payable_account_id",
+        origin_id_column,
+    )
+    convert_from_company_dependent(
+        env,
+        "account.tax.group",
+        "property_advance_tax_payment_account_id",
+        "advance_tax_payment_account_id",
+        origin_id_column,
+    )
+
+
+def _force_install_account_payment_term_module_module(env):
+    """
+    Force install account_payment_term if we need
+    key 'days_end_of_month_on_the' of it
+    it has already merged in odoo master
+    """
+    account_payment_term_module = env["ir.module.module"].search(
+        [("name", "=", "account_payment_term")]
+    )
+    needs_account_payment_term = bool(
+        env["account.payment.term.line"].search(
+            [("delay_type", "=", "days_end_of_month_on_the")], limit=1
+        )
+    )
+    if needs_account_payment_term and account_payment_term_module:
+        account_payment_term_module.state = "to install"
+        openupgrade.copy_columns(
+            env.cr,
+            {
+                "account_payment_term_line": [
+                    ("days_after", "days_next_month", "CHARACTER VARYING")
+                ]
+            },
+        )
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE account_payment_term_line
+            SET
+            nb_days = nb_days - days_after,
+            days_next_month = days_after
+            WHERE delay_type = 'days_end_of_month_on_the'
+            """,
+        )
+
+
+def _map_chart_template_id_to_chart_template(
+    env, model_table, coa_m2o="chart_template_id", coa_name_column="chart_template"
+):
+    """
+    In tables that used to refer to a chart template in the database, set the new
+    column identitying the COA by the name of its localization module with the
+    `l10n_` prefix removed (usually the country's iso code)
+    """
+    env.cr.execute(
+        f"""SELECT m.id, CONCAT(imd.module, '.', imd.name)
+            FROM {model_table} m
+                JOIN ir_model_data imd
+                    ON imd.model='account.chart.template'
+                        AND m.{coa_m2o} = imd.res_id
+            WHERE m.{coa_m2o} IS NOT NULL
+            AND CONCAT(imd.module, '.', imd.name) IN %s
+        """,
+        (tuple(COA_MAPPING),),
+    )
+    chart_id2name = [(_id, COA_MAPPING[xmlid]) for _id, xmlid in env.cr.fetchall()]
+    openupgrade.map_values(
+        env.cr,
+        coa_m2o,
+        coa_name_column,
+        chart_id2name,
+        table=model_table,
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    _account_payment_term_migration(env)
+    _account_payment_term_early_payment_discount(env)
+    _force_install_account_payment_term_module_module(env)
+    openupgrade.load_data(env, "account", "17.0.1.2/noupdate_changes.xml")
```

Comment:
after this, call delete_record_translations for email_template_edi_credit_note, email_template_edi_invoice and mail_template_data_payment_receipt